RecoEgamma/EgammaElectronAlgos/interface/EgammaPixelMatchParamObjects.h

```diff
@@ -209,9 +323,26 @@ namespace egPM {
       if(funcType.first=="TF1" && has1D<ParamType>(0)) return TF1Wrap<ParamType,has1D<ParamType>(0)>(funcType.second,funcParams);
       else if(funcType.first=="TF2" && has2D<ParamType>(0)) return TF2Wrap<ParamType,has2D<ParamType>(0)>(funcType.second,funcParams);
       else if(funcType.first=="TF3" && has3D<ParamType>(0)) return TF3Wrap<ParamType,has3D<ParamType>(0)>(funcType.second,funcParams);
-      else throw cms::Exception("InvalidConfig") << " type "<<funcType.first<<" is not recognised or is imcompatable with the ParamType, configuration is invalid and needs to be fixed"<<std::endl;
+      else throw cms::Exception("InvalidConfig") << " type "<<funcType.first<<" is not recognised or is imcompatable with the ParamType, configuration is invalid and needs to be fixed "<<std::endl;
+    }
+  };
+
+  //a constant bin of 0th dimension, ignores the input and just returns a const value
+  template<typename InputType>
+  class ParamBin0D : public ParamBin<InputType> {
+  private:
+    float val_;
+  public: 
+    ParamBin0D(const edm::ParameterSet& config):
+      val_(config.getParameter<double>("val")){} 
+    bool pass(const InputType&)const override {
+      return true;
+    }
+    float operator()(const InputType& input)const override{
+      return val_;
     }
   };
+    
 
   template<typename InputType,typename ParamType>
   class ParamBin1D : public ParamBin<InputType> {
```

Comment:
for the current implementation
```C++
     float operator()(const InputType& input)const override{
       if(!pass(input)) return 0;
       else return func_(ParamType(input));
     }
```

it looks like in many (all?) cases, this operator is called right after ```pass(input)``` call returns true.
It would be good to extend the interface or add some other logic to not call ```pass(input)``` again